components/create_post/create_post.jsx

```diff
@@ -415,35 +468,42 @@ export default class CreatePost extends React.Component {
     }
 
     componentWillMount() {
```

Comment:
I suggests to have these lifecycle methods to move at top location, just after the constructor and before other functions.